### ISSUE TITLE
Plots is saved in wrong path when `root.dir` is set 

### ISSUE THREAD

**Issue by cderv** [rstudio/reticulate#1526]
From a report at 
* https://github.com/quarto-dev/quarto-cli/issues/8460

### Example

````markdown
---
title: "reticulate"
output:
  html_document:
    keep_md: true
---

```{r}
library(knitr)
library(reticulate)
knitr::opts_knit$set(root.dir = 'data/')
```

```{r}
x <- 1:5
plot(x, x)
```

```{python}
import matplotlib.pyplot as plt
x = [1,2,3,4,5]
plt.plot(x, x)
```

````

The plot from python will be created in `data/` while the fig path included in the intermediate .md is relative to document directory, as expected and happening with R chunk. 

````markdown
```r
x <- 1:5
plot(x, x)
```

![](test_files/figure-html/unnamed-chunk-2-1.png)<!-- -->


```python
import matplotlib.pyplot as plt
x = [1,2,3,4,5]
plt.plot(x, x)
```

<img src="test_files/figure-html/unnamed-chunk-3-1.png" width="614" />

````

So the image is not found in the HTML file. 

### Past discussions

This is also related to past discussions at 
* https://github.com/rstudio/reticulate/issues/645#issuecomment-566616817
* https://github.com/rstudio/reticulate/pull/671
* https://github.com/yihui/knitr/issues/1783

I believe this was fixed in 1.15 version by #671 but it was lost in refactoring at https://github.com/rstudio/reticulate/commit/db9bb673948a45ff89847004a66328606c4938bb

## The problem

- **reticulate** engine operates in working directory set to `root.dir` knitr options, as this is how **knitr** calls any engine. 
- when computing the figure path, **reticulate** correctly take into account the output dir as where to create the file 
	https://github.com/rstudio/reticulate/blob/0e5676e0f86d28a5d2a4f532debcb3568a59618d/R/knitr-engine.R#L397-L426
	it changes the working directory accordingly. 
	but the path returned is only related to this new working dir. 
- so when this figure path is used, the working directory has been reset, and the file path where it is saved is no more the absolute right one
	https://github.com/rstudio/reticulate/blob/0e5676e0f86d28a5d2a4f532debcb3568a59618d/R/knitr-engine.R#L428-L441

- This was correct in #671 fix because `plt$savefig(path, dpi = options$dpi)` was done in the same function where 
	```r
	owd <- setwd(dir)
	on.exit(setwd(owd), add = TRUE)
	```
	was done. 

### Test file about the situation 

Below a file content to `knitr::knit()` to show the difference in configuration with path option between R chunk and Python chunk 

<details>
<summary>Input file </summary>

````markdown
---
title: "reticulate"
---

Change the knit root directory to the temporary R session.

```{r setup}
knitr::opts_knit$set("root.dir" = tempdir())
```

```{r confirm-wd}
getwd()
```

```{r configure-python}
library("reticulate")
matplotlib <- import("matplotlib")
```

The R plot is saved relative to the location of the Rmd file:

```{r r-plot}
plot(1:10)
```

The Python plot is saved relative to the working directory (`root.dir`):

```{python py-plot}
import matplotlib.pyplot as plt

x = range(1, 10)
plt.plot(x, x)
```

```{r}
knitr::opts_knit$get("root.dir")
file.exists(file.path(knitr::opts_knit$get("root.dir"), "figure", "r-plot-1.png"))
file.exists(file.path(knitr::opts_knit$get("root.dir"), "figure", "py-plot-1.png"))
file.exists(file.path(knitr::opts_knit$get("output.dir"), "figure", "r-plot-1.png"))
file.exists(file.path(knitr::opts_knit$get("output.dir"), "figure", "py-plot-1.png"))
```

````

</details>

Using `knitr::knit()` on it and it will produce the following

<details>
<summary>.md output</summary>

````markdown
---
title: "reticulate"
---



Change the knit root directory to the temporary R session.


```r
knitr::opts_knit$set("root.dir" = tempdir())
```


```r
getwd()
```

```
## [1] "C:/Users/chris/AppData/Local/Temp/Rtmp23FSnY"
```


```r
library("reticulate")
matplotlib <- import("matplotlib")
```

The R plot is saved relative to the location of the Rmd file:


```r
plot(1:10)
```

![plot of chunk r-plot](figure/r-plot-1.png)

The Python plot is saved relative to the working directory (`root.dir`):


```python
import matplotlib.pyplot as plt

x = range(1, 10)
plt.plot(x, x)
```

<div class="figure">
<img src="figure/py-plot-1.png" alt="plot of chunk py-plot" width="672" />
<p class="caption">plot of chunk py-plot</p>
</div>


```r
knitr::opts_knit$get("root.dir")
```

```
## [1] "C:\\Users\\chris\\AppData\\Local\\Temp\\Rtmp23FSnY"
```

```r
file.exists(file.path(knitr::opts_knit$get("root.dir"), "figure", "r-plot-1.png"))
```

```
## [1] FALSE
```

```r
file.exists(file.path(knitr::opts_knit$get("root.dir"), "figure", "py-plot-1.png"))
```

```
## [1] TRUE
```

```r
file.exists(file.path(knitr::opts_knit$get("output.dir"), "figure", "r-plot-1.png"))
```

```
## [1] TRUE
```

```r
file.exists(file.path(knitr::opts_knit$get("output.dir"), "figure", "py-plot-1.png"))
```

```
## [1] FALSE
```
````

</details>

Hope it is clear. Happy to help if needed.

**Comment by t-kalinowski** [comment]
@cderv Thank you for the detailed writeup - very helpful!

I have a tentative fix in PR #1529. Can you please try it out and confirm it fixes the issue?

```r
remotes::install_github("rstudio/reticulate#1529")
```

**Comment by t-kalinowski** [comment]
@cderv Thank you for the detailed writeup - very helpful!

I have a tentative fix in PR #1529. Can you please try it out and confirm it fixes the issue?

```r
remotes::install_github("rstudio/reticulate#1529")
```

**Comment by cderv** [comment]
After doing a `pak::pak("rstudio/reticulate#1529")`, I can confirm that the plots shows using R Markdown, and using Quarto 

Thanks a lot.
